src/stan/mcmc/chainset.hpp

```diff
@@ -0,0 +1,486 @@
+   * rank-normalized ESS for both bulk and tail ESS.
+   *
+   * @param index parameter index
+   * @return pair (bulk_ess, tail_ess)
+   */
+  std::pair<double, double> split_rank_normalized_ess(const int index) const {
+    return analyze::split_rank_normalized_ess(samples(index));
+  }
+
+  /**
+   * Computes the effective sample size (ESS) for the specified
+   * parameter across all chains, according to the algorithm presented in
+   * https://arxiv.org/abs/1903.08008, section 3.2 for folded (split)
+   * rank-normalized ESS for both bulk and tail ESS.
+   *
+   * @param name parameter name
+   * @return pair (bulk_ess, tail_ess)
+   */
+  std::pair<double, double> split_rank_normalized_ess(
+      const std::string& name) const {
+    return split_rank_normalized_ess(index(name));
+  }
+
+  /**
+   * Computes the mean Monte Carlo error estimate for the central 90% interval.
+   * See https://arxiv.org/abs/1903.08008, section 4.4.
+   * Follows implementation in the R posterior package
+   *
+   * @param index parameter index
+   * @return mcse
+   */
+  double mcse_mean(const int index) const {
+    return analyze::mcse_mean(samples(index));
+  }
+
+  /**
+   * Computes the mean Monte Carlo error estimate for the central 90% interval.
+   * See https://arxiv.org/abs/1903.08008, section 4.4.
+   * Follows implementation in the R posterior package.
+   *
+   * @param name parameter name
+   * @return mcse
+   */
+  double mcse_mean(const std::string& name) const {
+    return mcse_mean(index(name));
+  }
+
+  /**
+   * Computes the standard deviation of the Monte Carlo error estimate
+   * https://arxiv.org/abs/1903.08008, section 4.4.
+   * Follows implementation in the R posterior package.
+   *
+   * @param index parameter index
+   * @return mcse_sd
+   */
+  double mcse_sd(const int index) const {
+    return analyze::mcse_sd(samples(index));
+  }
+
+  /**
+   * Computes the standard deviation of the Monte Carlo error estimate
+   * https://arxiv.org/abs/1903.08008, section 4.4.
+   * Follows implementation in the R posterior package
+   *
+   * @param name parameter name
+   * @return mcse_sd
+   */
+  double mcse_sd(const std::string& name) const { return mcse_sd(index(name)); }
+
+  /**
+   * Compute autocorrelation for one column of one chain.
+   * Throws exception if column index is out of bounds.
+   * Autocorrelation is computed using Stan math library implmentation.
+   *
+   * @param chain chain index
+   * @param index column index
+   * @return vector of chain autocorrelation at all lags
+   */
+  Eigen::VectorXd autocorrelation(const int chain, const int index) const {
+    Eigen::MatrixXd s = samples(index);
+    Eigen::Map<const Eigen::VectorXd> chain_col(samples(chain).data(),
+                                                num_samples());
+    Eigen::VectorXd autocorr_col(num_samples());
+    stan::math::autocorrelation<double>(s.col(chain), autocorr_col);
```

Comment:
Q: If ess and friends are using stan::analyze::autocorrelation, does it matter which one we use here?

I'm not sure if this is actually being called any more, so maybe not?

Reply:
this matches what's being done in `chains.hpp`:  https://github.com/stan-dev/stan/blob/2550bbd2bbde86352b9cf9e1bc520968e3a88788/src/stan/mcmc/chains.hpp#L173-L188